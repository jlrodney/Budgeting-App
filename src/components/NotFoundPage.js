import React from 'react';
import { Link } from 'react-router-dom';

const NotFoundPage = () => (
  <div>
    404 Page not found - <Link to="/">Go home</Link>
  </div>
);

export default NotFoundPage;
