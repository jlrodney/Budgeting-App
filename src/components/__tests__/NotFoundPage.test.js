import React from 'react';
import { shallow, mount } from 'enzyme';
import {MemoryRouter, Link} from 'react-router';
import NotFoundPage from '../NotFoundPage';

describe('NotFoundPage should render  correctly', () => {
  it('should have a div', () => {
    const wrapper = shallow(<NotFoundPage />);
    expect(wrapper.find('div').text()).toEqual("404 - <Link />");
  })
  it('should have a link to homepage', () => {
    const wrapper = shallow(<NotFoundPage />);
    expect(wrapper.find('Link').prop('to')).toBe("/");
  })
});

describe('should render Header correctly', () => {
  const wrapper = shallow(<NotFoundPage />);
  expect(wrapper).toMatchSnapshot();
})
