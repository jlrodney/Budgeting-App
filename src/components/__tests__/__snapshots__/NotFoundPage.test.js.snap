// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div>
  404 Page not found - 
  <Link
    replace={false}
    to="/"
  >
    Go home
  </Link>
</div>
`;
